packages/docgen/src/compile.js

```diff
@@ -0,0 +1,47 @@
+const ts = require( 'typescript' );
+const { readFileSync } = require( 'fs' );
+
+const { getExportStatements } = require( './get-export-statements' );
+
+/**
+ * Function that takes file path and returns compiled result
+ *
+ * @param {string} filePath the path to the file to be compiled
+ */
+module.exports = function( filePath ) {
+	const options = {
+		allowJs: true,
+	};
+	const host = ts.createCompilerHost( options, true );
+	const raw = readFileSync( filePath ).toString();
+	const code = raw
+		// typescript interprets @wordpress in @example code as a JSDoc tag.
+		// So, it should be replaced for the time being.
+		// They're restored in get-jsdoc-from-statement.js
+		.replace( /@wordpress/g, '__WORDPRESS_IMPORT__' )
+		// When <caption>ES(5|Next)<\/caption> exists next to @example tag,
+		// typescript cannot parse code correctly.
+		// So, they're removed.
+		.replace( /@example\s+<caption>ES(5|Next)<\/caption>/g, '@example' );
+	const sF = ts.createSourceFile( filePath, code, options.target, true );
```

Comment:
I think it's reasonably self-explanatory based on the function name from which it's being assigned, but in general I would always encourage naming variables in their full, non-abbreviated form (i.e. `sourceFile`), since abbreviations can cause unnecessary difficulty in interpreting for future maintenance. We can afford the few extra characters 😄

Reply:
I named it in that way, because there was a `sourceFile` below to be exported. I haven't read the TypeScript code, but I guess that they're same. 

But to make it clear, I changed it from `sF` to `internalSourceFile`.